vizro-core/docs/pages/user_guides/table.md

```diff
@@ -0,0 +1,286 @@
+                      color: white
+                    - if:
+                        filter_query: "{lifeExp} < 55"
+                        column_id: lifeExp
+                      backgroundColor: "#85144b"
+                      color: white
+                    - if:
+                        filter_query: "{gdpPercap} > 10000"
+                        column_id: gdpPercap
+                      backgroundColor: green
+                      color: white
+                    - if:
+                        column_type: text
+                      textAlign: left
+                    - if:
+                        state: active
+                      backgroundColor: rgba(0, 116, 217, 0.3)
+                      border: 1px solid rgb(0, 116, 217)
+                id: table
+                type: table
+            controls:
+              - column: continent
+                type: filter
+            title: Dash DataTable
+
+        ```
+    === "Result"
+        [![Table2]][Table2]
+
+    [Table2]: ../../assets/user_guides/table/styled_table.png
+
+#### Custom Table
+
+In case you want to add custom logic to a Dash DataTable, e.g. when requiring computations that can be controlled by parameters, it is possible to
+create a custom Dash DataTable in Vizro.
+
+For this, similar to how one would create a [custom chart](../user_guides/custom_charts.md), simply do the following:
+
+- define a function that returns a  `dash_table.DataTable` object
+- decorate it with the `@capture("table")` decorator
+- the function must accept a `data_frame` argument (of type `pandas.DataFrame`)
+- the table should be derived from and require only one `pandas.DataFrame` (e.g. any further dataframes added through other arguments will not react to dashboard components such as `Filter`)
+
+
+The following example shows a possible version of a custom table. In this case the argument `chosen_columns` was added, which you can control with a parameter:
+
+??? example "Custom Dash DataTable"
+    === "app.py"
+        ```py
+        from typing import List
+
+        from dash import dash_table
+
+        import vizro.models as vm
+        import vizro.plotly.express as px
+        from vizro import Vizro
+        from vizro.models.types import capture
+
+        df = px.data.gapminder().query("year == 2007")
+
+
+        @capture("table")
+        def my_custom_table(data_frame=None, chosen_columns: List[str] = None):
+            """Custom table."""
+            columns = [{"name": i, "id": i} for i in chosen_columns]
+            defaults = {
+                "style_as_list_view": True,
+                "style_data": {"border_bottom": "1px solid var(--border-subtle-alpha-01)", "height": "40px"},
+                "style_header": {
+                    "border_bottom": "1px solid var(--state-overlays-selected-hover)",
+                    "border_top": "1px solid var(--main-container-bg-color)",
+                    "height": "32px",
+                },
+            }
```

Comment:
Agree! I was hesitant to do this, but otherwise the table would have looked out of sync with the other examples, so I decided to had it. The focus here is more on the custom argument, less on "generality"